aten/src/ATen/native/sparse/SparseMatMul.cpp

```diff
@@ -191,25 +207,18 @@ void sparse_matmul_kernel(
   */
 
   auto M = mat1.size(0);
-  auto K = mat1.size(1);
   auto N = mat2.size(1);
 
-  auto mat1_indices_ = mat1._indices().contiguous();
-  auto mat1_values = mat1._values().contiguous();
-  Tensor mat1_row_indices = mat1_indices_.select(0, 0);
-  Tensor mat1_col_indices = mat1_indices_.select(0, 1);
-
-  Tensor mat1_indptr = coo_to_csr(mat1_row_indices.data_ptr<int64_t>(), M, mat1._nnz());
-
-  auto mat2_indices_ = mat2._indices().contiguous();
-  auto mat2_values = mat2._values().contiguous();
-  Tensor mat2_row_indices = mat2_indices_.select(0, 0);
-  Tensor mat2_col_indices = mat2_indices_.select(0, 1);
-
-  Tensor mat2_indptr = coo_to_csr(mat2_row_indices.data_ptr<int64_t>(), K, mat2._nnz());
+  const auto mat1_csr = mat1.to_sparse_csr();
+  const auto mat2_csr = mat2.to_sparse_csr();
 
-  auto nnz = _csr_matmult_maxnnz(M, N, mat1_indptr.data_ptr<int64_t>(), mat1_col_indices.data_ptr<int64_t>(),
-      mat2_indptr.data_ptr<int64_t>(), mat2_col_indices.data_ptr<int64_t>());
+  const auto nnz = _csr_matmult_maxnnz(
+      M,
+      N,
+      mat1_csr.crow_indices().data_ptr<int64_t>(),
+      mat1_csr.col_indices().data_ptr<int64_t>(),
+      mat2_csr.crow_indices().data_ptr<int64_t>(),
+      mat2_csr.col_indices().data_ptr<int64_t>());
 
```

Comment:
@pearu , is this always safe? COO is int64 only, right?

Reply:
yes, COO indices are always int64